Introduce drag and drop support

kblok opened the pull request:
closes #1744

kgar commented:
I'm glad to review this. It'll take me a little time to read and understand.

kgar commented on lib/PuppeteerSharp.Tests/DragAndDropTests/InputDragTests.cs:
```diff
@@ -0,0 +1,132 @@
+            Assert.True(Page.IsDragInterceptionEnabled);
+            var draggable = await Page.QuerySelectorAsync("#drag");
+            var dropzone = await Page.QuerySelectorAsync("#drop");
+            await draggable.DragAndDropAsync(dropzone);
```
🔥 this is the stuff

kgar commented on lib/PuppeteerSharp/Page.cs:
```diff
@@ -2529,9 +2538,8 @@ public void Dispose()
         /// <param name="disposing">Indicates whether disposal was initiated by <see cref="Dispose()"/> operation.</param>
         protected virtual void Dispose(bool disposing) => _ = DisposeAsync();
-        #endregion
 
```
I recommend removing a newline to satisfy the Style Cop rule for newlines.

kblok replied:
done!

kgar commented on lib/PuppeteerSharp/Input/Mouse.cs:
```diff
@@ -149,5 +149,114 @@ public Task WheelAsync(decimal deltaX, decimal deltaY)
+        /// <param name="endY">End Y coordinate</param>
+        /// <param name="delay">If specified, is the time to wait between `dragover` and `drop` in milliseconds.</param>
+        /// <returns></returns>
+        /// <exception cref="System.NotImplementedException"></exception>
```
Is this `NotImplementedException` tag still needed in the xml doc?

kblok replied:
No. Removed :)

kgar commented on lib/PuppeteerSharp/Page.cs:
```diff
@@ -369,6 +367,11 @@ public int DefaultTimeout
+        /// <summary>
+        /// `true` if drag events are being intercepted, `false` otherwise.
+        /// </summary>
+        public bool IsDragInterceptionEnabled { get; internal set; }
```
I recommend making the setter `private set;`

kblok replied:
done!

kgar reviewed (changes requested):
Disclaimer: This is my first ever review on Github. Thank you for letting me be a part of this.

I have a few suggestions in the comments.
Of those suggestions, I get an analyzer-based compilation error because of the additional newline on `Page.cs`.

kblok commented:
>Disclaimer: This is my first ever review on Github. Thank you for letting me be a part of this.
>I have a few suggestions in the comments.
>Of those suggestions, I get an analyzer-based compilation error because of the additional newline on Page.cs.

You're welcome! It's always good to get some fresh eyes. I addressed all your comments.

Thanks

kgar reviewed (approved):
🎉🎉🎉
